packages/flutter_tools/lib/src/commands/init.dart

```diff
@@ -30,17 +31,30 @@ class InitCommand extends Command {
       return 2;
     }
 
+    if (ArtifactStore.flutterRoot == null) {
+      print('Neither the --flutter-root command line flag nor the FLUTTER_ROOT environment');
```

Comment:
You can use `stderr.writeln` here (and below) instead of `print`, as we're about to fail.